configure.ac

```diff
@@ -515,6 +515,11 @@ AC_ARG_ENABLE(thread-cs,
                          threads are present.  Values may be global
                          (default), per-object, per-vci, lock-free]),,enable_thread_cs=global)
 
+AC_ARG_ENABLE(vci-method,
```

Comment:
Do we need these hashing options? We could also rely on the other standard comm info hints to decide the hashing mechanism? E.g., if MPI_NO_ANY_SOURCE is present, we can use tag based hashing. 

If we were to use these vci-methods, which method will consider those comm hints?

Reply:
The hashing scheme that considers MPI_NO_ANY_SOURCE etc. can be a separate `vci-method`.

Reply:
The only relevant options seems `ZERO` and `IMPLICIT`

Reply:
@rzambre your benchmarks will work with "COMM". My experiment with legion will work with "TAG", Intel's internal branch will with "IMPLICIT", and the endpoint proposal will work with "EXPLICIT". We need all these config options available for testing purposes before we settle down to the final solutions.

Reply:
From my perspective, `COMM` and `IMPLICIT` are the same. If the user doesn't give any hints, then it is the same as `COMM`. And, `TAG` is incorrect. So, I wouldn't prefer they go into `master`. IMO, testing with hacky approaches is for branches in forks. I'm not sure what the guidelines are for things going into MPICH `master`. So, I'll let you be the judge.

Reply:
Well, there could be multiple "implicit" schemes, before it settles down, we'll need multiple config options for each schemes that we are interested in. So for now, "IMPLICIT" does not mean all "implicit" schemes, it refers to "one" implicit scheme (the one in Intel's internal branch). 

I think "master" is allowed to have experimental features as long as it does not interfere or contaminate production configurations. Here removing or adding a hashing scheme is trivia and each scheme is well isolated from each other, so I think it is OK.